test/e2e/variation_index.test.js

```diff
@@ -0,0 +1,126 @@
+const algoliasearch = require('algoliasearch');
+const { execSync } = require('child_process');
+
+describe('Algolia Integration', () => {
+    beforeAll(async () => {
+        if (!process.env.CIRCLECI) {
+            console.log('Skipping Algolia Integration tests - not in CircleCI environment');
+            return;
+        }
+
+        // First verify SFCC auth credentials exist
+        const sfccAuthVars = ['SFCC_OAUTH_CLIENT_ID', 'SFCC_OAUTH_CLIENT_SECRET'];
+        sfccAuthVars.forEach(envVar => {
+            if (!process.env[envVar]) {
+                throw new Error(`Missing SFCC authentication variable: ${envVar}`);
+            }
+        });
+
+        // Authenticate with SFCC using explicit environment variables
+        try {
+            execSync(
+                `sfcc-ci client:auth "${process.env.SFCC_OAUTH_CLIENT_ID}" "${process.env.SFCC_OAUTH_CLIENT_SECRET}"`,
```

Comment:
As discussed, `sfcc-ci` has a JavaScript API, any reason no to use it? https://github.com/SalesforceCommerceCloud/sfcc-ci?tab=readme-ov-file#using-the-javascript-api